### PR TITLE
REFACTOR: Added status flags

### PR DESCRIPTION
**Pull request by coachchucksol**
- Added status flags for better tracking of the state machine
- Fixed issue on new epoch not being able to reset state to idle

### CODE REVIEW

File: programs/steward/src/state/steward_state.rs [hunk 1]
```diff
@@ -236,7 +230,37 @@ impl IdlBuild for StewardStateEnum {
...
+pub const COMPUTE_DELEGATIONS: u32 = 1 << 1;
+pub const EPOCH_MAINTENANCE: u32 = 1 << 2;
+pub const PRE_LOOP_IDLE: u32 = 1 << 3;
+pub const COMPUTE_INSTANT_UNSTAKES: u32 = 1 << 4;
+pub const REBALANCE: u32 = 1 << 5;
+pub const POST_LOOP_IDLE: u32 = 1 << 6;
+// BITS 8-15 RESERVED FOR FUTURE USE
+// BITS 16-23 OPERATIONAL FLAGS
+pub const CHECKED_VALIDATORS_REMOVED_FROM_LIST: u32 = 1 << 16;
+pub const RESET_TO_IDLE: u32 = 1 << 17;
```
Comment (ebatsell) [inline programs/steward/src/state/steward_state.rs:244]: could we make this an enum so the other files don't have to import all these constants?
```rust
pub enum StatusFlags {
  ComputeScore = COMPUTE_SCORE,
  ComputeDelegations = ...
}

File: programs/steward/src/state/steward_state.rs [hunk 2]
```diff
@@ -383,25 +412,24 @@ impl StewardState {
...
             self.state_tag = StewardStateEnum::Idle;
-            self.rebalance_completed = true.into();
+            self.set_flag(REBALANCE);
         }
         Ok(())
     }
 
     /// Update internal state when transitioning to a new cycle, and ComputeScores restarts
-    fn reset_state_for_new_cycle(
+    pub fn reset_state_for_new_cycle(
```
Comment (ebatsell) [inline programs/steward/src/state/steward_state.rs:432]: remove this pub?

File: programs/steward/src/state/steward_state.rs [hunk 3]
```diff
@@ -110,24 +110,18 @@ pub struct StewardState {
     /// Total lamports that have been due to stake deposits this cycle
     pub stake_deposit_unstake_total: u64,
 
+    /// Various flags to track state progress - See
```
Comment (ebatsell) [inline programs/steward/src/state/steward_state.rs:113]: unfinished comment

File: programs/steward/src/state/steward_state.rs [hunk 4]
```diff
@@ -383,25 +412,24 @@ impl StewardState {
         num_epochs_between_scoring: u64,
     ) -> Result<()> {
         if current_epoch >= self.next_cycle_epoch {
-            self.state_tag = StewardStateEnum::ComputeScores;
```
Comment (ebatsell) [inline programs/steward/src/state/steward_state.rs:386]: what's the reason for removing all of these?

File: programs/steward/src/state/steward_state.rs (hunk 1 above)
Comment (ebatsell) [inline programs/steward/src/state/steward_state.rs:244]: can you add comments explaining these 2 operational statuses?

Comment (coachchucksol) [reply programs/steward/src/state/steward_state.rs:244]: It all started out with an enum - but I fought Rust allowing me to go between u32 and the enum. I had all the functions and everything compiling, then I decided it was overly verbose for something that should be simple. That being said, I can change it back into an enum if you'd like.

File: programs/steward/src/state/steward_state.rs (hunk 2 above)
Comment (coachchucksol) [reply programs/steward/src/state/steward_state.rs:432]: 10-4 that was an artifact from a different solution

File: programs/steward/src/state/steward_state.rs (hunk 3 above)
Comment (coachchucksol) [reply programs/steward/src/state/steward_state.rs:113]: 10-4

File: programs/steward/src/state/steward_state.rs [hunk 5]
```diff
@@ -236,7 +230,45 @@ impl IdlBuild for StewardStateEnum {
     }
 }
 
+// BITS 0-7 COMPLETED PROGRESS FLAGS
+// Used to mark the completion of a particular state
+pub const COMPUTE_SCORE: u32 = 1 << 0;
+pub const COMPUTE_DELEGATIONS: u32 = 1 << 1;
+pub const EPOCH_MAINTENANCE: u32 = 1 << 2;
+pub const PRE_LOOP_IDLE: u32 = 1 << 3;
+pub const COMPUTE_INSTANT_UNSTAKES: u32 = 1 << 4;
```
Comment (ebatsell) [inline programs/steward/src/state/steward_state.rs:239]: nit: COMPUTE_INSTANT_UNSTAKE to be aligned with the instruction name

**Pull request merged**